bootstrap/bootstrap-api/src/main/java/com/alipay/sofa/rpc/bootstrap/DefaultConsumerBootstrap.java

```diff
@@ -262,14 +260,12 @@ public List<ProviderGroup> subscribe() {
         String directUrl = consumerConfig.getDirectUrl();
         if (StringUtils.isNotEmpty(directUrl)) {
             // 如果走直连
-            result = subscribeFromDirectUrl(directUrl);
-        } else {
-            // 没有配置url直连
-            List<RegistryConfig> registryConfigs = consumerConfig.getRegistry();
-            if (CommonUtils.isNotEmpty(registryConfigs)) {
-                // 从多个注册中心订阅服务列表
-                result = subscribeFromRegistries();
-            }
+            consumerConfig.setRegistry(new RegistryConfig().setProtocol(REGISTRY_PROTOCOL_DOMAIN));
+        }
+        List<RegistryConfig> registryConfigs = consumerConfig.getRegistry();
```

Comment:
If directurl is set, there should be no need to subscribe from other registries.